### PR TITLE
Add command to test coverage for java tests

### LINKED ISSUE #43: Add a make rule to test coverage of the java tests
We should be able to report our coverage. Add a make rule `make coverage` that automatically runs the tests, and generates a coverage report

### PR DESCRIPTION
**Pull request by rehnarama**
Fixes #43 

Simply run the command `make coverage`.

This will produce the folder `report/` which has the file `index.html`. Open it with your browser of choice: this holds all coverage date.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,14 @@
-.PHONY: test
+.PHONY: test compile_test coverage
 
 SUPPORT_LIBRARY = support_libraries
 JAVA_ROOT = build/java
 
-test:
+compile_test:
```
Comment (JonasNorlinder) [inline Makefile]: Looking on what it does it is just an alias for compiling, and not really have anything to do with tests.
```suggestion
compile
```

File: Makefile [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
 	ant java
+
+test: compile_test
```
Comment (JonasNorlinder) [inline Makefile]:
```suggestion
test: compile
```

File: Makefile [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
 	java -cp ".:${SUPPORT_LIBRARY}/hamcrest-core-1.3.jar:${SUPPORT_LIBRARY}/junit-4.13-beta-3.jar:${JAVA_ROOT}" org.junit.runner.JUnitCore org.python.types.ListTest
+
+coverage: compile_test
```
Comment (JonasNorlinder) [inline Makefile]:
```suggestion
coverage: compile
```

File: Makefile [hunk 4]
```diff
@@ -1,8 +1,14 @@
-.PHONY: test
+.PHONY: test compile_test coverage
```
Comment (JonasNorlinder) [inline Makefile]: (see comment below)
```suggestion
 .PHONY: test compile coverage
```

**Review by JonasNorlinder — CHANGES_REQUESTED** [review]
Great! 🎈 Just a small suggestion regarding the naming to reflect its actual meaning.

File: .gitignore [hunk 5]
```diff
@@ -21,4 +21,7 @@ tests/tmp*
 
 .DS_Store
 .idea/
-.vscode
\ No newline at end of file
+.vscode
```
Comment (Melkster) [inline .gitignore]: Adding editor-specific patterns in `.gitignore` is generally considered bad practice. See [this article](https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore) on how to create a global `.gitignore` for your machine.

**Pull request merged**